### ISSUE TITLE
CA2000 does not trigger for ServiceProvider

### ISSUE THREAD

**Issue by sorensenmatias** [dotnet/roslyn-analyzers#6152]
### Analyzer

**Diagnostic ID**: [CA2000](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.400](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

CA2000 does not trigger when creating an instance of `Microsoft.Extensions.DependencyInjection.ServiceProvider`, even though `ServiceProvider` implements `IDisposable`.

### Steps To Reproduce

Create new `csproj` file with the following content:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="6.0.0" />
  </ItemGroup>

</Project>
```

Create new `.editorconfig` file in the same directory as the `csproj` file.
Configure CA2000 as warning in `.editorconfig`:

```
[*]

dotnet_diagnostic.CA2000.severity = warning
```

Add the following class in a new file:

```
public class MyClass
{
    public void MyMethod()
    {
        var serviceCollection = new ServiceCollection();
        var serviceProvider = serviceCollection.BuildServiceProvider();
    }
}
```

### Expected behavior

CA2000 suggests to add the missing using-statement, i.e.:

```
using var serviceProvider = serviceCollection.BuildServiceProvider();
```

### Actual behavior

CA2000 does not trigger.

**Comment by sorensenmatias** [comment]
I figured the cause of this issue is the same as mentioned here: https://github.com/dotnet/roslyn-analyzers/issues/5330#issuecomment-928889891
Basically, the instance of `ServiceProvider` is created in a referenced assembly, and CA2000 does not recognize the ownership of the `IDisposable` as being transferred to our code.

I see the following options as ways to solve this:

1) Enhance `IsDisposableCreationSpecialCase` so it also considers methods starting with "build".

```
 private static bool IsDisposableCreationSpecialCase(IMethodSymbol targetMethod) 
     => targetMethod.IsStatic && 
        (targetMethod.Name.StartsWith("create", StringComparison.OrdinalIgnoreCase) || 
         targetMethod.Name.StartsWith("open", StringComparison.OrdinalIgnoreCase) ||
         targetMethod.Name.StartsWith("build", StringComparison.OrdinalIgnoreCase) ||
         ); 
```

2) Create option for assuming that ownership of any `IDisposable` is transferred when it is returned from a method. This can potentially give a lot of noise for consumers. See https://github.com/dotnet/roslyn-analyzers/issues/1587 for related options.

3) Await design of Disposable annotations to make its way. With this, the `BuildServiceProvider` method could be annotated as transferring the ownership, and CA2000 would pick this up. See https://github.com/dotnet/roslyn-analyzers/issues/1617 and
https://github.com/dotnet/runtime/issues/29631. Probably a long wait for this.
